src/html_support_files/odoc_search.js

```diff
@@ -64,3 +64,91 @@ document.querySelector(".search-bar").addEventListener("input", (ev) => {
   wait();
   worker.postMessage(ev.target.value);
 });
+
+
+/** Navigation */
+
+let search_result_elt = document.querySelector(".search-result")
+
+function search_results() {
+  return search_result_elt.children;
+}
+
+function enter_search() {
+  document.querySelector(".search-bar").focus();
+}
+
+function escape_search() {
+  document.activeElement.blur()
+}
+
+function focus_previous_result() {
+  let results = Array.from(search_results());
+  let current_focus = results.findIndex((elt) => (document.activeElement === elt));
+  if (current_focus === -1)
+    return;
+  else if (current_focus === 0)
+    enter_search();
+  else
+    results[current_focus - 1].focus();
+}
+
+function focus_next_result() {
+  let results = Array.from(search_results());
+  if (results.length === 0) return;
+  let current_focus = results.findIndex((elt) => (document.activeElement === elt));
+  if (current_focus === -1)
+    results[0].focus();
+  else if (current_focus + 1 === results.length)
+    return;
+  else
+    results[current_focus + 1].focus();
+}
+
+function enter_focus() {
+  if (n_focus === null) return;
+  let elt = current_result();
+  elt.click();
+}
+
+function is_searching() {
+  return (document.querySelectorAll(".odoc-search:focus-within").length > 0);
+}
+
+function is_typing() {
+  return (document.activeElement === document.querySelector(".search-bar"));
+}
+
+function handle_key_down(event) {
+  if (is_searching()) {
+    if (event.key === "ArrowUp") {
+      event.preventDefault();
+      focus_previous_result();
+    }
+    if (event.key === "ArrowDown") {
+      event.preventDefault();
+      focus_next_result();
+    }
+    if (event.key === "Escape") {
+      event.preventDefault();
+      escape_search();
+    }
+  }
+  if (!(is_typing())) {
+    let ascii = event.key.charCodeAt(0);
+    if (event.key === "/") {
+      event.preventDefault();
+      enter_search();
+    }
+    else if ( event.key.length === 1
+            && (ascii >= 65 && ascii <= 90) // lowercase letter
+            || (ascii >= 97 && ascii <= 122) // uppercase letter
+            || (ascii >= 48 && ascii <= 57) // numbers
+            || (ascii >= 32 && ascii <= 46) // ` ` to `.`
+            || (ascii >= 58 && ascii <= 64) // `:` to `@`
```

Comment:
I'm not sure we want that all those ascii characters trigger search...

Reply:
> > (nitpick) It would be nice if the focus were teleported to the input when typing again:
> This is done as well

Ah, maybe this was due to this! But typing on any ascii character even outside of search triggers search, maybe that is too much?